File: src/main/java/net/hectus/neobb/turn/default_game/block/TRespawnAnchor.java
package net.hectus.neobb.turn.default_game.block;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.buff.ChancedBuff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.HotClazz;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import org.bukkit.block.Block;

import java.util.List;

public class TRespawnAnchor extends BlockTurn implements BuffFunction, HotClazz {
    public TRespawnAnchor(NeoPlayer player) { super(player); }
    public TRespawnAnchor(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public boolean requiresUsageGuide() { return true; }

    @Override
    public int cost() {
        return 4;
    }

    @Override
    public List<Buff> buffs() {
        return List.of(new Buff.Teleport(Buff.BuffTarget.YOU), new ChancedBuff(new Buff.ExtraTurn(), 50));
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/block/TRespawnAnchor.kt
package net.hectus.neobb.modes.turn.default_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.buff.ChancedBuff
import net.hectus.neobb.buff.ExtraTurn
import net.hectus.neobb.buff.Teleport
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.HotClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.BuffFunction
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.util.asCord
import org.bukkit.block.Block

class TRespawnAnchor(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), BuffFunction, HotClazz {
    override val cost: Int = 4

    override fun buffs(): List<Buff<*>> {
        return listOf(Teleport(player!!.game.history.lastOrNull()?.location()?.asCord() ?: Cord(0.0, 0.0, 0.0)), ChancedBuff(50.0, ExtraTurn()))
    }
}
